File: Implementation/python/11_단어뒤집기2.py
# fast campus 강의
# https://www.acmicpc.net/problem/17413
# 1

S = input()
temp, result, ck = "", "", False

for s in S:
    if s == ' ':
        if not ck:
            result += temp[::-1] + ' '
            temp = ""
        else: result += ' '
    elif s == '<':
        ck = True
        result += temp[::-1] + '<'
        temp = ""
    elif s == '>':
        ck = False
        result += '>'
    else:
        if ck: result += s
        else: temp += s

result += temp[::-1]
print(result)

File: backup/Implementation/c++/11_단어뒤집기2.cpp
// fast campus 강의
// https://www.acmicpc.net/problem/17413
// 2

#include <iostream>
#include <algorithm>
#include <sstream>

using namespace std;

int main()
{
	string s, result, tmp;
	getline(cin, s);

	bool tag_chk = false;
	for (int i = 0; i < s.size(); i++) {
		if (s[i] == ' ') {
			if (tag_chk)
				result += s[i];
			else {
				reverse(tmp.begin(), tmp.end());
				result += tmp + ' ';
				tmp = "";
			}
		}
		else if (s[i] == '>') {
			result += '>';
			tag_chk = false;
		}
		else if (s[i] == '<') {
			reverse(tmp.begin(), tmp.end());
			result += tmp + '<';
			tmp = "";
			tag_chk = true;
		}
		else {
			if (tag_chk) result += s[i];
			else tmp += s[i];
		}
	}
	reverse(tmp.begin(), tmp.end());
	result += tmp;
	cout << result;
}